pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -729,3 +739,12 @@ func buildCatchAllVirtualHost(node *model.Proxy) *route.VirtualHost {
 		IncludeRequestAttemptCount: true,
 	}
 }
+
+// Simply removes everything before .svc, if present
+func removeSvcNamespace(domain string) string {
+	if idx := strings.Index(domain, ".svc."); idx > 0 {
+		return domain[idx:]
```

Comment:
nit: I think this is redundant check for the context in which this is being used.

Reply:
It's actually possible for a hostname here not to have `.svc` in it (in which case, just returning the domain unchanged is the right thing to do).  I fixed the comments to make that more clear.   Sorry!

Reply:
Ha, yes.  Ignore my original response - I had confused domain with hostname in my comment.  It is true that `domain` will always have `.svc`. As it turns out, though, despite the check not being necessary, the `gocritic` check/bot took issue with the fact that result could be `-1`.  So I think I have to keep the check in if I use `strings.Index` here at all?